Add build command sanity check to close #3502

sudosays opened the pull request:
## Description of the Pull Request (PR):

Addresses #3502
Modified the `checkBuildTarget` function to first try and parse the target as a def file if build target already exists and force overwrite flag is false. This method is insensitive to filenaming and is based on the content of the file itself.

### This fixes or addresses the following GitHub issues:

 - Closes #3502

Attn: @singularity-maintainers

sudosays committed:
Add build command sanity check to close #3502

Sanity check involves parsing the build target if it exists and checking
whether or not it is a valid definition file.

dtrudg commented on cmd/internal/cli/build.go:
```diff
@@ -265,7 +265,17 @@ func checkBuildTarget(path string) error {
+
+			isDefFile, _ := parser.IsValidDefinition(path)
+			if isDefFile {
+				question = "Build target is a valid definition file that will be overwritten. Do you still want to overwrite? [N/y]"
```
I think `valid` is unnecessary here

dtrudg reviewed (commented):
Thanks @sudosays !

While looking at this, I wondered what if we make this even more explicit, and put the basename of the target file in the question messages in both cases? That should maximally highlight to the user what they are about to overwrite.

`Build target 'mydef.def' is a definition file that will be overwritten. Do you still want to overwrite? [N/y]`

`Build target 'myoldsif.sif' already exists. Do you want to overwrite? [N/y]`

sudosays committed:
Add file base name to sanity check message

Sanity check still differentiates between definition files and others,
but now gives the file name in both cases.

dtrudg commented on cmd/internal/cli/build.go:
```diff
@@ -265,7 +265,17 @@ func checkBuildTarget(path string) error {
 		if !buildArgs.update && !forceOverwrite {
-			question := "Build target already exists. Do you want to overwrite? [N/y] "
+
+			var question string
```
Apologies - I should have made this comment earlier. Please let me know if you'd rather I commit this change to the branch, rather than you needing to do it, and I can take care of that to merge this.


It would probably be more common to write something like this...

```
question := fmt.Sprintf("Build target '%s' already exists. Do you want to overwrite? [N/y]", f.Name())

if isDefFile, _ := parser.IsValidDefinition(path); isDefFile {
    question = fmt.Sprintf("Build target '%s' is a definition file that will be overwritten. Do you still want to overwrite? [N/y]", f.Name())	
}
```

The pattern of setting a default value with `:=` and overriding for the special case, rather than using `var` and `if/else` blocks is generally used in our code, and in Go code in general.

sudosays replied:
I have fixed it on my side with the new commit. Thanks for the advice, I'll make sure to conform to that style in future.

sudosays committed:
Fix variable assignment in build sanity check

The assignment of the question variable has been changed to align more
with the code standards for the project and Golang.
